Gems/PhysX/Code/Tests/RigidBodyComponentTests.cpp

```diff
@@ -140,4 +140,52 @@ namespace PhysXEditorTests
         // Check the geometry is a type of Convex
         EXPECT_EQ(pxShape->getGeometryType(), physx::PxGeometryType::eCONVEXMESH);
     }
+
+    TEST_F(PhysXEditorFixture, EditorRigidBodyComponent_CylinderColliderZeroSize_NoColliderCreated)
+    {
+        // Create editor entities
+        EntityPtr editorEntities[2] = { CreateInactiveEditorEntity("ZeroRadius"), CreateInactiveEditorEntity("ZeroHeight") };
+
+        constexpr const char* expectedErrors[2] = { "SetCylinderRadius: radius must be greater than zero.",
+                                                     "SetCylinderHeight: height must be greater than zero."};
+
+        for (int i = 0; i < 2; ++i)
+        {
+            UnitTest::ErrorHandler expectedError(expectedErrors[i]);
+            EntityPtr& editorEntity = editorEntities[i];
+
+            const auto* rigidBodyComponent = editorEntity->CreateComponent<PhysX::EditorRigidBodyComponent>();
+            const auto* colliderComponent = editorEntity->CreateComponent<PhysX::EditorColliderComponent>();
+
+            editorEntity->Activate();
+
+            AZ::EntityComponentIdPair idPair(editorEntity->GetId(), colliderComponent->GetId());
+
+            // Set collider to be a cylinder
+            const Physics::ShapeType shapeType = Physics::ShapeType::Cylinder;
+            PhysX::EditorColliderComponentRequestBus::Event(idPair, &PhysX::EditorColliderComponentRequests::SetShapeType, shapeType);
+
+            // Set collider cylinder radius and height
+            if (i == 0)
+            {
+                PhysX::EditorColliderComponentRequestBus::Event(
+                    idPair, &PhysX::EditorColliderComponentRequests::SetCylinderRadius, 0.0f);
+            }
+            else
+            {
+                PhysX::EditorColliderComponentRequestBus::Event(
+                    idPair, &PhysX::EditorColliderComponentRequests::SetCylinderHeight, 0.0f);
+            }
```

Comment:
this looks really opaque I would probably just split this into two test cases. more code but the setup and assertions would be more cleaer.

Reply:
Yeah I did consider having two test cases initially but thought its testing pretty much the same functionality and it wasn't enough to justify extra code. 
I agree having separate test cases is more "correct" here, so changed now, thanks!